pytorch_lightning/utilities/hparams_mixin.py

```diff
@@ -0,0 +1,180 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
on directory structure, this is under utilities/hparams_mixin but is used by core/lightning. does it make sense for this to be part of core too? should we have a `core/mixins` subdirectory?

Reply:
On board with that, could move `device_dtype_mixin` & `hparams_mixin` to a separate directory in `core` in a follow up PR.

Reply:
Yes, I like this idea too. Especially if we move to a more opt-in structure for the Lightning module.

Reply:
coming back to the root, I thought we want to away from mixins
cc: @awaelchli @justusschock

Reply:
my perspective has always been: we want to move away from these multiple-inheritance abuses in the trainer that are NOT mixins (can name the classes mixins as much as we want, they are not). 

the class introduced here is a true mixin, and it's use is justified IMO.